### PR TITLE
Bound chat session fields before allocation

### PR DESCRIPTION
**Pull request by kdcokenny**
I was checking chat session update decoding and noticed packet-controlled key and signature lengths were cast from `i32` to `usize` and used for allocation before validation.

A negative or oversized VarInt could therefore request an enormous allocation from a tiny packet.

Fixed it by matching vanilla's 512-byte public-key and 4096-byte signature limits and using bounded prefixed reads before allocation. Also added tests for boundary, malformed, and truncated packets.

Part of #328

### CODE REVIEW

**Review by JunkyDeveloper — APPROVED** [review]

File: steel-protocol/src/packets/game/chat/session.rs [hunk 1]
```diff
@@ -91,13 +95,9 @@ impl steel_utils::serial::ReadFrom for SChatSessionUpdate {
...
-        let key_len = VarInt::read(reader)?.0 as usize;
-        let mut public_key = vec![0u8; key_len];
-        reader.read_exact(&mut public_key)?;
-
-        let sig_len = VarInt::read(reader)?.0 as usize;
-        let mut key_signature = vec![0u8; sig_len];
-        reader.read_exact(&mut key_signature)?;
+        let public_key = Vec::<u8>::read_prefixed_bound::<VarInt>(reader, MAX_PUBLIC_KEY_SIZE)?;
+        let key_signature =
+            Vec::<u8>::read_prefixed_bound::<VarInt>(reader, MAX_KEY_SIGNATURE_SIZE)?;
```
Comment (coco875) [inline steel-protocol/src/packets/game/chat/session.rs:100]: a random though now, we can just make fixed size array instead of vec ? to allocate on the stack instead of the heap ?

Comment (kdcokenny) [reply steel-protocol/src/packets/game/chat/session.rs:100]: These are variable-length fields; 512 and 4096 are only their maximums. Fixed arrays would need separate lengths and would inflate every ScheduledPlayPacketKind value by roughly 4ish KiB. The bounded Vec match the Vanilla impl and only allocates the actual wire length.

Comment (coco875) [reply steel-protocol/src/packets/game/chat/session.rs:100]: I though it can like C https://godbolt.org/z/evbbP51Er, didn't know that it wasn't possible in rust

Comment (kdcokenny) [reply steel-protocol/src/packets/game/chat/session.rs:100]: Ahh, you mean a C variable-length array. It could work during decoding, but it dies when the function returns, and this packet is queued for later, so we’d still need to copy it into owned storage. Due to this, i'd think a bounded Vec fits better here.

**Review by coco875 — APPROVED** [review]
